src/server/middleware/v2auth/auth.go

```diff
@@ -98,16 +113,24 @@ func getAction(req *http.Request) rbac.Action {
 
 }
 
-var checker = reqChecker{
-	pm: config.GlobalProjectMgr,
-}
+var (
+	once    sync.Once
+	checker reqChecker
+)
 
 // Middleware checks the permission of the request to access the artifact
 func Middleware() func(http.Handler) http.Handler {
+	once.Do(func() {
+		if checker.pm == nil { // for UT, where pm has been set to a mock value
+			checker = reqChecker{
+				pm: config.GlobalProjectMgr,
+			}
+		}
+	})
```

Comment:
Changing the value of the `checker` at the beginning of the UT doesn't work ? If it doesn't work, the right place of `once.Do` is  in `check` method of `reqChecker` not in the middleware.

Reply:
The problem is that config.GlobalProjectMgr is nil when checker is assigned.  So when this middleware was added to the handler I found it didn't work.